torch/ao/quantization/qconfig.py

```diff
@@ -249,7 +249,7 @@ def assert_valid_qconfig(qconfig: Optional[Union[QConfig, QConfigDynamic]],
     """
     Verifies that this `qconfig` is valid.
     """
-    if qconfig is None:
+    if qconfig is None or qconfig.weight is None:
```

Comment:
might be cleaner to move the weight check inside the if statement, something like

```
if is_conv_transpose_mod:
    if qconfig.weight is None:
      # for now, we assume that any qconfig for ConvTranspose without a weight is valid
      return 
    ...
```